### PR TITLE
Update packaging to handle regional ad block files

### PR DESCRIPTION
**Pull request by emerick**

### CODE REVIEW

File: manifests/ad-block-updater-FD176DD1-F9A0-4469-B43E-B1764893DD5C-manifest.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "description": "Brave Ad Block Updater extension",
+  "key": "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAoqpe6QWKketr66AW+hKSxr5qds9gxmU72N20bG/nG8wfbPUdTwEqIG3m5e37Iq1FI0gcQir5UqwZZUgkum5dWJwgCB1SOaVvlZrWZbTbATKFcePswHqAIXMfS+wzMA/Ifoky26dE4k/rs6J/zQEbeXXon/ikjGJ7GxYeHYMBz9DAPQhcUoBlh1C0O0vhvXU+kG5DO4wdIt9W/cXJtv+8OTZ6HiTJw1j0jAliFZI/jhkYB6MW57OBpBYlWJQhMbLbK5opXq6d4ELbjC1amqI1lT3j5bl0g1OpMqL4Jtz6578G79gMJfxE3hA5tL0rGU3vAmwck/jXh7uOOzqetwdBcwIDAQAB",
+  "manifest_version": 2,
+  "name": "Brave Ad Block Updater (BGR: Bulgarian Adblock list)",
+  "version": "0.0.0"
+}
```
Comment (bbondy) [inline manifests/ad-block-updater-FD176DD1-F9A0-4469-B43E-B1764893DD5C-manifest.json]: Can we use the ad-block node native lib to generate these automatically from the ad-block list, or are you still having trouble with getting it built as a dep?

Comment (emerick) [reply manifests/ad-block-updater-FD176DD1-F9A0-4469-B43E-B1764893DD5C-manifest.json]: Good idea, I'll look into it. Haven't tried your tip yet about `GYP_MSVS_VERSION`, I'll give that a shot!

Comment (emerick) [reply manifests/ad-block-updater-FD176DD1-F9A0-4469-B43E-B1764893DD5C-manifest.json]: Fixed in newest commit.

File: scripts/packageComponent.js [hunk 2]
```diff
@@ -10,62 +11,130 @@ const replace = require('replace-in-file')
...
+function getDATFileListByComponentType (componentType) {
+  let list = []
+
+  switch (componentType) {
+    case 'ad-block-updater':
+      fs.readdirSync(path.join('node_modules', 'ad-block', 'out'))
+        .filter(file => {
+          return (path.extname(file) === '.dat' && file !== 'SafeBrowsingData.dat')
+        })
+        .forEach(file => {
```
Comment (bbondy) [inline scripts/packageComponent.js]: I'm ok with this but just noting you could do a reduce after the filter to achieve the same slightly more cleanly.

File: scripts/packageComponent.js [hunk 3]
```diff
@@ -10,62 +11,130 @@ const replace = require('replace-in-file')
...
-const datFile = getDATFileFromComponentType(commander.type)
-const crxFile = path.join(outputDir, commander.type + '.crx')
-const privateKeyFile = commander.key
+generateManifestFilesByComponentType(commander.type)
 
-stageFiles(commander, outputDir, datFile)
-generateCRXFile(crxFile, privateKeyFile, inputDir, outputDir)
+getDATFileListByComponentType(commander.type).forEach(datFile => {
+  processDATFile(commander.type, keyParam, datFile)
+})
```
Comment (bbondy) [inline scripts/packageComponent.js]: I think you could do
`getDATFileListByComponentType(commander.type).forEach(processDATFile.bind(null, commander.type, keyParam))`

File: scripts/packageComponent.js [hunk 4]
```diff
@@ -10,62 +11,130 @@ const replace = require('replace-in-file')
...
+}
+
+function processDATFile (componentType, key, datFile) {
+  const datFileName = getNormalizedDATFileName(path.parse(datFile).name)
+  const stagingDir = path.join('build', componentType, datFileName)
+  const manifestsDir = getManifestsDirByComponentType(componentType)
+  const crxOutputDir = path.join('build', componentType)
+  const crxFile = path.join(crxOutputDir, `${componentType}-${datFileName}.crx`)
+
+  let privateKeyFile = ''
```
Comment (bbondy) [inline scripts/packageComponent.js]: could simplify with 

```const privateKeyFile = !fs.lstatSync(key).isDirectory() ? key : path.join(key, `${componentType}-${datFileName}.pem`)```

File: scripts/packageComponent.js [hunk 5]
```diff
@@ -10,62 +11,130 @@ const replace = require('replace-in-file')
...
+      return path.join('node_modules', 'ad-block', 'out')
+    case 'tracking-protection-updater':
+      // TODO(emerick): Make this work like ad-block
+      return path.join('manifests')
+    default:
+      throw new Error('Unrecognized component extension type: ' + componentType)
+  }
+}
+
+function getNormalizedDATFileName (datFileName) {
```
Comment (bbondy) [inline scripts/packageComponent.js]: could simplify to:

```const getNormalizedDATFileName = (datFileName) => datFileName === 'ABPFilterParserData' || datFileName === 'TrackingProtection' ? 'default' : datFileName```

**Comment by emerick** [comment]
@bbondy Incorporated code review feedback, can you review when you have a few spare cycles? Thanks!

**Pull request merged**